ext/semian/resource.c

```diff
@@ -34,6 +40,11 @@ semian_resource_acquire(int argc, VALUE *argv, VALUE self)
   TypedData_Get_Struct(self, semian_resource_t, &semian_resource_type, self_res);
   res = *self_res;
 
+  if (res.quota > 0) {
+    // Ensure that configured tickets match the quota before acquiring
+    update_tickets_from_quota(res.sem_id, res.quota);
```

Comment:
oh, here it's letting me reply, weird.

Yes, this is because workers die the registered workers will be out of since with the configured workers.